### PR TITLE
Feature/rnd 120 add env to moto

### PR DESCRIPTION
**Pull request by oriroza**

### CODE REVIEW

**Review by barakber — COMMENTED** [review]
- I would squash all commits to a single one  
- We don't need the test here  
- We don't need any prints here

File: moto/core/models.py [hunk 1]
```diff
@@ -788,6 +789,27 @@ def __call__(self, func=None):
...
             return mocked_backend
 
+ecs_backends_global = {}
+class atidot_base_decorator(base_decorator):
+    mock_backend = MockAWS
+
+    def __init__(self, backends):
+        super().__init__(backends)
+
+    def __call__(self, func=None, callback=None):
```
Comment (barakber) [inline moto/core/models.py:798]: I don't think we need this implementation at all (of `__call__`)

File: moto/core/models.py [hunk 2]
```diff
@@ -788,6 +789,27 @@ def __call__(self, func=None):
         else:
             return mocked_backend
 
+ecs_backends_global = {}
+class atidot_base_decorator(base_decorator):
+    mock_backend = MockAWS
+
+    def __init__(self, backends):
+        super().__init__(backends)
```
Comment (barakber) [inline moto/core/models.py:796]: do this here -
~~~python
     for backend in self.backends:
            self.backends[backend].at_callback = callback
            # why not?
            # backend.at_callback = callback
~~~

File: tests/test_ecs/test_aaa.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
+from __future__ import unicode_literals
```
Comment (barakber) [inline tests/test_ecs/test_aaa.py]: Remove this file

File: moto/ecs/models.py [hunk 4]
```diff
@@ -1587,11 +1600,12 @@ def update_service_primary_task_set(self, cluster, service, primary_task_set):
...
 
-
-ecs_backends = {}
-for region in Session().get_available_regions("ecs"):
-    ecs_backends[region] = EC2ContainerServiceBackend(region)
-for region in Session().get_available_regions("ecs", partition_name="aws-us-gov"):
-    ecs_backends[region] = EC2ContainerServiceBackend(region)
-for region in Session().get_available_regions("ecs", partition_name="aws-cn"):
-    ecs_backends[region] = EC2ContainerServiceBackend(region)
+def get_ecs_backends():
```
Comment (barakber) [inline moto/ecs/models.py]: Why are you messing up with `backends`?

File: moto/ecs/models.py [hunk 5]
```diff
@@ -774,6 +778,15 @@ def run_task(self, cluster_str, task_definition_str, count, overrides, started_b
                         container_instance, resource_requirements
                     )
                     tasks.append(task)
+                    if self.at_callback:
+                        print("executing callback")
```
Comment (barakber) [inline moto/ecs/models.py]: don't print here (only in callback)

File: moto/ecs/models.py [hunk 6]
```diff
@@ -3,6 +3,7 @@
 import re
 import uuid
 from copy import copy
+from subprocess import Popen, PIPE
```
Comment (barakber) [inline moto/ecs/models.py]: remove

File: moto/core/models.py [hunk 7]
```diff
@@ -2,6 +2,7 @@
 from __future__ import unicode_literals
 from __future__ import absolute_import
 
+import copy
```
Comment (barakber) [inline moto/core/models.py]: remove

File: moto/core/models.py (hunk 1 above)
Comment (oriroza) [reply moto/core/models.py:798]: @barakber  that's the place which we get the callback from decorator

File: moto/core/models.py (hunk 2 above)
Comment (oriroza) [reply moto/core/models.py:796]: backend is a name of region...(str type)

**Pull request closed without merging**